### DISCUSSION TITLE
Is the authentication tutorial out of date?

### DISCUSSION THREAD

**Discussion by JohnGarrison78** [serverpod/serverpod#3022] [Q&A] [answered]
I'm trying to follow the tutorial here: https://medium.com/serverpod/getting-started-with-serverpod-authentication-part-1-72c25280e6e9 and it doesn't work at all. It's linked from the official docs, so that's concerning. The first time I tried to create an account, the validation code didn't work. But somehow, the account was created anyway. Any subsequent attempts to create an account tell me that the email is already in use. I was very excited about using Dart for both the backend and the front end, but this has deflated that excitement. I hope there's some updated documentation somewhere with some steps that are missing from the one I tried. If it helps, I'm using a Neon hosted postgres instead of the Docker one. That is setup correctly, and Serverpod added data to it for the one user that I was able to halfway create. 

EDIT: I also tried resetting the password for the user that never got validated. That validation code is also rejected.

**Comment by SandPod** [comment]
Hey @JohnGarrison78!

Yes, unfortunately, this guide has not been updated to Serverpod 2.0, but the steps should still be compatible.

Let's try to figure out what might have gone wrong, and we can patch the guide to ensure this does not happen to other people.

When creating new user accounts through email authentication, the authentication module utilizes three tables.

First an entry is created in the `serverpod_email_create_request` table. This holds placeholder information for the user that will be created at a later stage when the email is validated.

Once the email has been verified using the validation code, the entry in the `serverpod_email_create_request` is removed, and an entry in the `serverpod_user_info` along with `serverpod_email_auth` is created. These are used to store the authentication credentials and general user information.

In what table in the database do you see the created user?

**Neon** - Using Neon for postgres should not be an issue.

**Comment by JohnGarrison78** [comment]
serverpod_email_create_request has the ones I tried to create that wouldn't even let me enter the validation code. Those gave me the error that the email already existed, even when it was my first time entering that email. I think one of the github issues says that error goes out regardless of the actual error. I don't know if that has been fixed. 

The one that let me enter the validation code and told me it was the wrong one was removed from serverpod_email_create_request. It exists in the other two, however, it told me that the validation failed. Sadly, I tried again so many times I don't remember the original password. So I don't know if that account works at all. I can't create new ones and I can't reset the password because of the validation code error there.

**Reply by SandPod** [reply, marked as answer]
You can reset the validation state by simply removing the entries in the database.

I know that Neon provides a great interface for editing rows in the database. It should be simple to remove any entry from  `serverpod_email_create_request` and `serverpod_email_auth` to start again.

The password is never retrievable from the database, so If you have forgotten your credentials I suggest you either reset the password or remove the user.

> I think one of the github issues says that error goes out regardless of the actual error.

This is true, when creating the account request the reason why the creation process was rejected is not communicated to the client but it is logged on the server. But if it was rejected there should not have been an entry added to the `serverpod_email_create_request` table. So if you see an entry there, the account request was successfully added.
